enhancements/installer/assisted-installer.md

```diff
@@ -0,0 +1,330 @@
+     hostname of the bootstrap host in advance of building the
+     cluster.
+   * We could ask the user to enter a URL or other similar information
+     on the console when each host boots the live ISO, directing that
+     host to the current bootstrap host.
+   * We could ask the user to enter the IPs of hosts into the UI they
+     use to drive the installation and have the installer connect to
+     the agent on those IPs to instruct them to join the cluster. That
+     means the connection direction would be different for
+     disconnected than connected, where the agent reaches up to the
+     service and the service may not be able to reach the agent.
+
+## Summary
+
+This enhancement describes changes in and around the installer to
+assist with deployment on user-provisioned infrastructure. The use
+cases are especially relevant for bare metal, but will be applicable
+to cloud users who are running the installer directly instead of using
+a front-end such as `cloud.redhat.com` or the UI provided by their cloud
+vendor.
+
+## Motivation
+
+The target user is someone wanting to deploy OpenShift, especially on
+bare metal, with as little up-front infrastructure dependencies as
+possible. This person has access to servers and wants to run workloads
+quickly. They do not necessarily have the administrative privileges to
+create private VLANs, configure DHCP/PXE servers, or manage other
+aspects of the infrastructure surrounding the hardware where the
+cluster will run. They are willing to accept that the cluster they
+build may not have all of the same features as an IPI cluster on day
+1, but that by taking additional steps later they will be able to make
+it fully production-ready.
+
+### Goals
+
+- Make initial deployment of usable and supportable clusters simpler.
+- Move more infrastructure configuration from day 1 to day 2.
+- Support connected and disconnected on-premise deployments.
+- Support existing IPI features, especially for day 2 cluster
+  management and scale-out.
+
+### Non-Goals
+
+- Because the initial focus is for bare metal, this enhancement does
+  not exhaustively cover variations needed to offer similar features
+  on other platforms (such as changes to image formats, the way a host
+  boots, etc.). It is desirable to support those platforms, but that
+  work will be described separately.
+- Replace the existing OpenShift installer.
+- Describe how these workflows would work for multi-cluster
+  deployments managed with Hive or ACM.
+
+## Proposal
+
+There are several separate changes to enable the assisted installer
+workflows, including a GUI front-end for the installer, a cloud-based
+orchestration service, and changes to the installer and bootstrapping
+process.
+
+### Connected Deployments
+
+The process starts when the user goes to an "assisted installer"
+application running on `cloud.redhat.com`, enters details needed by
+the installer (pull secret, OpenShift version, ssh keys, proxy
```

Comment:
The pull secret can contain access keys to multiple registries, so in the disconnected case you need some way to provide those.

In dev-scripts we use `podman login` to convert a username/password into an authfile, which can then be merged with the OpenShift pull-secret to enable access to the local registry.

https://github.com/openshift-metal3/dev-scripts/blob/master/02_configure_host.sh#L195

Reply:
Also I guess this depends how we deliver the agent - it could be delivered as a container that's pulled to the regular installer iso, in which case you'd always need a pull secret to start the install process?

Reply:
We need to resolve this, but I don't think this draft doc is the right place for it. I added it as an open question so we don't lose track of the need to answer it.